### PR TITLE
Add sample code for bulk index request for Go client

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Add sample code for bulk index request for Go client

### Issues Resolved
Issue #507 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@VijayanB Please test to confirm sample code functions 

@kolchfa-aws Please test to perform QC of sample code

File: _clients/go.md [hunk 1]
```diff
@@ -116,6 +115,34 @@ func main() {
     }
     fmt.Println(searchResponse)
 
+    // Create the OpenSearch client
```
Comment (kolchfa-aws) [inline _clients/go.md:118]: Pls delete lines 118-133

File: _clients/go.md [hunk 2]
```diff
@@ -25,7 +24,7 @@ go get github.com/opensearch-project/opensearch-go
 
 ## Sample code
 
-This sample code creates a client, adds an index with non-default settings, inserts a document, searches for the document, deletes the document, and finally deletes the index:
+This sample code creates a client, adds an index with non-default settings, inserts a document, searches for the document, triggers a bulk index request, deletes the document, and deletes the index:
```
Comment (kolchfa-aws) [inline _clients/go.md:27]: Please change "triggers a bulk index request" to "performs bulk operations"

**Review by cwillum — APPROVED** [review]
LGTM

**Comment by kolchfa-aws** [comment]
The new PR that fixes this issue is #1038

**Pull request closed without merging**